app/controllers/tournaments_controller.rb

```diff
@@ -83,16 +83,22 @@ def save_pools
     @tournament = Tournament.includes(:pools, { pools: :pool_fighters }).find(params[:id])
     success = true
     message = "Pools saved"
-
     params[:pools].each do |key, pool|
       current_pool = @tournament.pools.find(pool[:id])
-      current_pool.pool_fighters.delete_all
-      current_fighters = []
-      pool[:fighters].try(:each) do |fighter|
-        current_fighters << { fighter_id: fighter }
+      current_pool_fighters = current_pool.fighters.pluck(:id) || []
```

Comment:
You don't need `|| []` here, since if you don't find anything, you'll get an empty array.